tests/crasher.c

```diff
@@ -87,6 +87,9 @@ write_maps(char *fname)
 #endif
 
 #ifdef __GNUC__
+// Gcc >= 8 became too good at inlining aliase c into b when using -O2 or -O3,
+// so force -O1 in all cases, otherwise a frame will be missing in the tests.
+#pragma GCC optimize "-O1"
```

Comment:
Not sure about the policy here but I believe this will cause a unknown pragma warning on clang.

Reply:
Do you want me to protect that with a #ifndef __clang__ ?

Reply:
That's the not sure about policy port ;-p

Another approach I've used that may be slightly more compiler friendly is to use a page allocated as `PROT_NONE`.

Reply:
I am not sure I understand the part with the page allocated as `PROT_NONE`.

Reply:
Well creating a segfault is not the problem here. It correctly segfaults.

Ok I realize I have badly posted my initial report. I should have added the first line:
```
Program received signal SIGSEGV, Segmentation fault.
```
as well for the first case. But it is really here.

The problem I am trying to fix in this pull request is not the segfault itself, but the unwinding part. Until now, gcc used not to inline the call to `c` (which is an alias of `b`) while now it does.

Reply:
Ah, sorry, I somehow thought you mean that the GCC optimized out the UB....... The clang warning should still be there though and I'm not the one to decide what's the best way for this project.
(Would be nice if the test/this test can just be always compilied with `-O1` or `-O0`.)

Reply:
> I should have added the first line:
>
> ```
> Program received signal SIGSEGV, Segmentation fault.
> ```
>
> as well for the first case

Ah, yeah, that's why I was confused..............